jax/_src/scipy/linalg.py

```diff
@@ -656,3 +656,13 @@ def sqrtm(A, blocksize=1):
   if blocksize > 1:
       raise NotImplementedError("Blocked version is not implemented yet.")
   return _sqrtm(A)
+
+@_wraps(scipy.linalg.hilbert)
+@partial(jit, static_argnames=("n",))
+def hilbert(n):
```

Comment:
It surprises me that there's no `dtype` option. I know scipy doesn't provide it, but given the computational characteristics of accelerators, maybe the JAX version should? What do you think?

Reply:
Good question. Would the `dtype` be passed on to `arange`, or would the resulting matrix be cast to the `dtype`, or both?

I am not as familiar with the computational characteristics of accelerators, but here are some thoughts:
* Other array-creation routines (e.g. https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.fft.fftfreq.html?highlight=fftfreq) don't do that, even though a similar case could be made here, too?
* Which types would even be relevant? Roughly those that are allowed by `arange` (Mostly `float`- and `complex`-style types)? 
* I am also wondering whether the `1/(a + aT)` computation might introduce some implicit type promotion that introduces subtle issues which would be hard to catch in tests. But I am just speculating...

So intuitively, I would not have added that option. Though I don't have a strong opinion, and if you think it is a good idea I will add a dtype argument :) Thoughts?

Reply:
`1/(a + aT)` would promote to inexact, so any inexact dtype should be fine.

We can probably punt on this now and add it if/when someone requests it.